src/prerecorded/options/mod.rs

```diff
@@ -0,0 +1,337 @@
+        self
+    }
+
+    pub fn keywords(mut self, keywords: impl IntoIterator<Item = &'a str>) -> Self {
+        self.keywords.extend(keywords);
+        self
+    }
+
+    pub fn utterances(mut self, utterances: Utterances) -> Self {
+        self.utterances = Some(utterances);
+        self
+    }
+
+    pub fn tag(mut self, tag: &'a str) -> Self {
+        self.tag = Some(tag);
+        self
+    }
+}
+
+impl<'a> Default for OptionsBuilder<'a> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Serialize for OptionsBuilder<'_> {
+    fn serialize<S>(&self, serializer: S) -> core::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut seq = serializer.serialize_seq(None)?;
+
+        // Destructuring it makes sure that we don't forget to use any of it
+        let Self {
+            model,
+            version,
+            language,
+            punctuate,
+            profanity_filter,
+            redact,
+            diarize,
+            ner,
+            multichannel,
+            alternatives,
+            numerals,
+            search,
+            callback,
+            keywords,
+            utterances,
+            tag,
+        } = self;
+
+        if let Some(model) = model {
+            let s = match model {
+                Model::General => "general",
+                Model::Meeting => "meeting",
+                Model::Phonecall => "phonecall",
+                Model::Voicemail => "voicemail",
+                Model::Finance => "finance",
+                Model::Conversational => "conversational",
+                Model::Video => "video",
+                Model::CustomId(id) => id,
+            };
+
+            seq.serialize_element(&("model", s))?;
+        }
+
+        if let Some(version) = version {
+            seq.serialize_element(&("version", version))?;
+        }
+
+        if let Some(language) = language {
+            let s = match language {
+                Language::zh_CN => "zh-CN",
+                Language::zh_TW => "zh-TW",
+                Language::nl => "nl",
+                Language::en_US => "en-US",
+                Language::en_AU => "en-AU",
+                Language::en_GB => "en-GB",
+                Language::en_IN => "en-IN",
+                Language::en_NZ => "en-NZ",
+                Language::fr => "fr",
+                Language::fr_CA => "fr-CA",
+                Language::de => "de",
+                Language::hi => "hi",
+                Language::id => "id",
+                Language::it => "it",
+                Language::ja => "ja",
+                Language::ko => "ko",
+                Language::pt => "pt",
+                Language::pr_BR => "pr_BR",
+                Language::ru => "ru",
+                Language::es => "es",
+                Language::es_419 => "es-419",
+                Language::sv => "sv",
+                Language::tr => "tr",
+                Language::uk => "uk",
+            };
```

Comment:
Consider putting this whole match block in an `impl Display for Language`

Reply:
I did try that at first, but `Display` is for building `String`s, not accessing `&str`s that already exist. It would work fine, but I didn't really want to add unnecessary overhead.

Is it worth doing anyways?

Reply:
generalizes nicely to the the `Other` case as well since it contains a `String`

Reply:
Sort of.  Display is for general formatting, which might involve creating a String, or might involve writing to a socket or a file using the `write!()` macro, which doesn't involve any allocation.

Reply:
Also, I would prioritize simplicity (specifically, not putting lifetimes on structs) in setup code, since that is never the bottleneck for performance.  (Especially when sending hundreds of megabytes of audio over the wire.

Reply:
> specifically, not putting lifetimes on structs

I agree with that, probably better for the `Other` variant to hold a `String` instead of a `&str`